eng/testing/linker/SupportFiles/Directory.Build.targets

```diff
@@ -29,6 +31,18 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="RootJavaScriptInteropAssembly"
```

Comment:
This should not be necessary with recent master

Reply:
I'll try again without it, but I was getting test failures due to this assembly getting trimmed for console apps.

Reply:
I just validated that with latest changes in dotnet/runtime, if I remove this target some tests start to fail. For example, [GenericArraySortHelperTest](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Runtime/tests/TrimmingTests/GenericArraySortHelperTest.cs) works fine with the target and returns the expected value, but without it I get the following:

```
root@168370dbdf5c:/home/joperezr/share/runtime/artifacts/bin/trimmingTests/projects/System.Runtime.TrimmingTests/GenericArraySortHelperTest/browser-wasm/bin/Release/net6.0/browser-wasm/AppBundle# ./run-v8.sh
Arguments: --run,project.dll
console.debug: MONO_WASM: Initializing mono runtime
console.debug: MONO_WASM: ICU data archive(s) loaded, disabling invariant mode
WASM-ERR: [ERROR] FATAL UNHANDLED EXCEPTION: Nested exception detected.
WASM-ERR: Original Exception: at System.Collections.Generic.Dictionary`2<string, object>..ctor (int,System.Collections.Generic.IEqualityComparer`1<string>) <0x00060>
WASM-ERR: at System.Collections.Generic.Dictionary`2<string, object>..ctor (int) <0x00014>
WASM-ERR: at System.AppContext.Setup (char**,char**,int) <0x00016>
WASM-ERR:
WASM-ERR: Nested exception:at System.RuntimeType.get_BaseType () <0x00056>
WASM-ERR: at System.Type.IsSubclassOf (System.Type) <0x00044>
WASM-ERR: at System.Reflection.CustomAttribute.IsDefined (System.Reflection.ICustomAttributeProvider,System.Type,bool) <0x00040>
WASM-ERR: at System.Reflection.RuntimeMethodInfo.IsDefined (System.Type,bool) <0x00016>
WASM-ERR: at System.Diagnostics.StackTrace.ShowInStackTrace (System.Reflection.MethodBase) <0x00048>
WASM-ERR: at System.Diagnostics.StackTrace.ToString (System.Diagnostics.StackTrace/TraceFormat,System.Text.StringBuilder) <0x000c8>
WASM-ERR: at System.Diagnostics.StackTrace.ToString (System.Diagnostics.StackTrace/TraceFormat) <0x00036>
WASM-ERR: at System.Exception.GetStackTrace (bool) <0x0006a>
WASM-ERR: at System.Exception.get_StackTrace () <0x0000e>
WASM-ERR: at System.Exception.ToString () <0x000ae>
WASM-ERR: at System.Exception.ToString () <0x00054>
WASM-ERR:
WASM-ERR:
```

Reply:
cc @lewing - as he was asking me about the same thing recently.

Do we have a coherent story for InteropServices.JavaScript in a wasm app?

Reply:
Then that's a bug we need to fix. The goal of these tests is to run out of the box with linker not to manually persist things here and there.

Reply:
@joperezr - can you log a follow up issue for this? I think the mono wasm team should look into it, since any wasm console app will have this same problem.

Reply:
https://github.com/dotnet/runtime/issues/48522

Reply:
Just to close the loop here, @tannergooding has found the underlying issue and posted PR https://github.com/dotnet/runtime/pull/48710 to fix it so I'll wait to merge this until that PR goes in and I'll remove the workaround from this PR.